### PR TITLE
plugins/transformations: Update error handling logic on successful transforms

### PR DESCRIPTION
**Pull request by nfuden**
# Description

Nillity check safety

# Context

Repeated calls with the same valid transform ran into a type conversion between nil interfaces.

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/6163

### CODE REVIEW

File: projects/gloo/pkg/plugins/transformation/plugin.go [hunk 1]
```diff
@@ -307,6 +307,8 @@ func (p *Plugin) validateTransformation(
 	if err, ok := p.validationLruCache.Get(transformHash); ok {
 		// Error may be nil here since it's just the cached result
 		return err.(error)
```
Comment (elcasteel) [inline projects/gloo/pkg/plugins/transformation/plugin.go:309]: do you need to delete this return?

Comment (nfuden) [reply projects/gloo/pkg/plugins/transformation/plugin.go:309]: LOLOL i was testing and forgot to remove good catch

**Comment by nfuden** [comment]
Updating for better testing per conversation

File: projects/gloo/pkg/plugins/transformation/plugin.go [hunk 2]
```diff
@@ -306,7 +306,9 @@ func (p *Plugin) validateTransformation(
 	// This transformation has already been validated, return the result
 	if err, ok := p.validationLruCache.Get(transformHash); ok {
 		// Error may be nil here since it's just the cached result
-		return err.(error)
+		// so return it as a nil err after cast worst case.
+		errCasted, _ := err.(error)
+		return errCasted
```
Comment (EItanya) [inline projects/gloo/pkg/plugins/transformation/plugin.go:311]: The other thing we can do is type cast error when putting it into the cache. Could be one extra line of protection.

**Review by EItanya — COMMENTED** [review]
Can we mock the validate function so we can also test errors?

File: projects/gloo/pkg/plugins/transformation/plugin.go [hunk 3]
```diff
@@ -61,7 +61,8 @@ type Plugin struct {
 	TranslateTransformation    TranslateTransformationFn
 	settings                   *v1.Settings
 
-	// validationLruCache is a map of: (transformation hash) -> (error state)
+	// validationLruCache is a map of: (transformation hash) -> error state
+	// where error state is of type interface{} on return
```
[the remark this answers is not in the document]
Comment (nfuden) [reply projects/gloo/pkg/plugins/transformation/plugin.go]: Fair point on the semantics there. How do you feel about the updated description?

**Review by sam-heilbron — COMMENTED** [review]
LGTM. I'd be in favor of casting the error before placing it in the cache, and then just returning the cached value, but this works as well

**Comment by Implausiblyfun** [comment]
It may not be crazy to cast prior to adding to the cache as well but from an ongoing safety perspective it probably doesn't hurt to keep this on access to future proof against other entries to the cache.
While making the input clean is good the conversion is quick and prevents panics.

**Comment by sam-heilbron** [comment]
Yeah that's fair. Given that, I think this is good to go as is

**Review by sam-heilbron — APPROVED** [review]

**Pull request merged**